### PR TITLE
Add camera and imu sensor 

### PR DESCRIPTION
**Pull request by D-Alex**

### CODE REVIEW

**Comment by doudou** [comment]
About the fixed joints: it's a good idea, but would have to be guarded with version ifdefs. Moreover, it will break the transformer integration (a.k.a. syskit) if you don't add support for it in the SDF transformer loader as well.

File: rock_gazebo.orogen [hunk 1]
```diff
@@ -81,3 +81,31 @@ task_context "LaserScanTask" do
...
+task_context "CameraTask" do
+    needs_configuration
+
+    subclasses 'BaseTask'
+
+    # Camera full name
+    attribute "name", "/std/string"
+
+    # Camera output port
+    output_port "frame", "/base/samples/frame/Frame"
```
Comment (doudou) [inline rock_gazebo.orogen]: ro_ptr(...)

File: rock_gazebo.orogen [hunk 2]
```diff
@@ -81,3 +81,31 @@ task_context "LaserScanTask" do
...
+
+    # IMU full name
+    attribute "name", "/std/string"
+    property('imu_frame', '/std/string', 'imu').
+    	doc 'the name of the frame attached to the IMU'
+    property('world_frame', '/std/string', 'imu_world').
+    	doc 'the name of the world frame'
+
+    output_port('orientation_samples', '/base/samples/RigidBodyState').
+        doc 'provides timestamped IMUReading samples containing the orientation estimate as reported by the IMU.'
```
Comment (doudou) [inline rock_gazebo.orogen]: Write documentation as comment above the element, e.g.

```
# Provides timestamped ...
output_port 'orientation_samples'
```

PS: the documentation string is obviously completely wrong ...

File: tasks/CameraTask.cpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+void CameraTask::cleanupHook()
+{
+    CameraTaskBase::cleanupHook();
+}
+
+void CameraTask::setGazeboModel( ModelPtr model, string sensorName, string topicName )
+{
+    string taskName = "gazebo:" + model->GetWorld()->GetName() + ":" + model->GetName() + ":" + sensorName;
+    if(!provides())
+        throw std::runtime_error("CameraTask::provides returned NULL");
```
Comment (doudou) [inline tasks/CameraTask.cpp:77]: Is that even possible ????

File: rock_gazebo.orogen (hunk 1 above)
Comment (D-Alex) [inline rock_gazebo.orogen]: You are right ;-). I will add it.

File: tasks/CameraTask.cpp (hunk 3 above)
Comment (D-Alex) [inline tasks/CameraTask.cpp:77]: Not sure, therefor I think it is fair to check it.

Comment (doudou) [inline tasks/CameraTask.cpp:77]: Actually, I am very sure it is NOT possible (just checked)

**Comment by D-Alex** [comment]
Is not that writing old data if the component was restarted?
https://github.com/D-Alex/simulation-orogen-rock_gazebo/blob/master/tasks/LaserScanTask.cpp#L52

**Comment by doudou** [comment]
> Is not that writing old data if the component was restarted?

Not only that, but it's probably writing the same sample over and over again as well.

**Comment by doudou** [comment]
Hey. I won't merge this until the transformer integration is adapted ...

**Comment by doudou** [comment]
> Hey. I won't merge this until the transformer integration is adapted ...

Wrong call on my side. If we start using fixed joints, we'll adapt the transformer SDF support for it. This is perfectly valid IMO.

**Pull request merged**